### ISSUE TITLE
Implement SpatialDropout1D (and dropout noise masks)

### ISSUE THREAD

**Issue by cpury** [tensorflow/tfjs#1033]
When loading a Keras model that uses the SpatialDropout1D layer, I get an error:

```
jquery-3.3.1.slim.min.js:2 Uncaught Error: Unknown layer: SpatialDropout1D. This may be due to one of the following reasons:
1. The layer is defined in Python, in which case it needs to be ported to TensorFlow.js or your JavaScript code.
2. The custom layer is defined in JavaScript, but is not registered properly with tf.serialization.registerClass().
```

It would be nice to have this kind of Dropout in TFJS (it's very important in NLP applications).

From my understanding, the same behavior can be achieved with the normal Dropout layer and choosing the right mask. In that case, it would be great if TFJS would understand this and convert it to a Dropout layer when loading.

**Comment by cpury** [comment]
Unfortunately, it seems Dropout with any non-default noise shape is not implemented yet: https://github.com/tensorflow/tfjs-layers/blob/master/src/layers/core.ts#L98

So that would be a first step...

**Comment by joaopedromattos** [comment]
Hey guys, any updates regarding this issue? I'm facing the same problem...
